mllib/src/main/scala/org/apache/spark/ml/clustering/KMeans.scala

```diff
@@ -237,6 +237,27 @@ class KMeans @Since("1.5.0") (
   @Since("1.5.0")
   def setSeed(value: Long): this.type = set(seed, value)
 
+  /** @group setParam */
+  @Since("2.0.0")
+  def setInitialModel(value: KMeansModel): this.type = set(initialModel, value)
+
+  /** @group setParam */
+  @Since("2.0.0")
+  def setInitialModel(value: Model[_]): this.type = {
```

Comment:
If this is something we intended to have be a general function, should probably go in the HasInitialModel trait.